### PR TITLE
Change secondary config to use Dart logging's Levels, and set default to INFO

### PR DESCRIPTION
**Pull request by gkc**
See #472 for context

* at_secondary_server/lib/src/server/at_secondary_config.dart
  * Removed debugLog and traceLog from AtSecondaryConfig
  * traceLog wasn't used anywhere. Usages of debugLog in main.dart and in secondary_config_test.dart - see comments below
  * Replaced with 'logLevel' which will come from env var 'logLevel' or from YAML log.level and defaults to INFO. AtSecondaryConfig.logLevel can never be null
  * Added a TODO to make everything else in AtSecondaryConfig null safe, since everything has default values
* at_secondary_server/bin/main.dart
  * Removed reference to `AtSecondaryConfig.debugLog`; now uses `AtSecondaryConfig.logLevel`
* secondary_config_test.dart
  * Removed check that AtSecondaryConfig.debugLog is a boolean
  * Added a test to ensure that log level will be initialized correctly
* at_secondary_server/config/config_production.properties
  * **Added `log.level=WARNING` as we want production secondaries log level to be 'WARNING', but the default in config-base.yaml remains as FINEST, equivalent to existing settings of debug=true**
    * @cpswan I believe this will have the desired effect but need some advice on how best to test it
* at_secondary_server/config/config-base.yaml
  * removed log.debug and log.trace, added log.level as 'FINEST' since log.debug and log.trace were previously true
* at_secondary_server/config/config.yaml
  * removed log.debug and log.trace, added log.level as 'FINEST' since log.debug and log.trace were previously true
* at_functional_test/lib/secondary/base/config/config.yaml
  * removed log.debug and log.trace, added log.level as FINEST since log.debug and log.trace were previously true

### CODE REVIEW

**Comment by cpswan** [comment]
@gkc we changed the way that we build and configure in staging a little while ago, so we no longer build a separate image carrying a separate config, but rather use ENV variables.

We also default to prod, so if we want WARNING in prod then that should be the default. (Though I think it should actually be INFO so that at least we get the version number into the logs at startup).

If we want the staging environment to log at FINEST then we should add that ENV variable to the Docker services there (and make it part of the template for future secondaries deployed there). Quite possibly we actually want staging to be as same as prod as possible, and we can selectively subject chosen secondaries to tailored logging levels.

**Comment by gkc** [comment]
Thanks @cpswan 


> @gkc we changed the way that we build and configure in staging a little while ago, so we no longer build a separate image carrying a separate config, but rather use ENV variables.

Does this mean that config-production.properties and generate-config.py are obsolete?

> We also default to prod, so if we want WARNING in prod then that should be the default.

Yeah I went this way because my assumption was that config-production.properties was being used for prod, so I therefore changed config-production.properties but tried to leave everything else as unchanged as possible. I'm a bit wary of changing default log levels for everyone's secondaries in their dev / testing 

> (Though I think it should actually be INFO so that at least we get the version number into the logs at startup).

OK

> If we want the staging environment to log at FINEST then we should add that ENV variable to the Docker services there (and make it part of the template for future secondaries deployed there).

OK - can you point me at the repo & file(s) where I can add that ENV variable?

> Quite possibly we actually want staging to be as same as prod as possible, and we can selectively subject chosen secondaries to tailored logging levels.

sure

**Comment by cpswan** [comment]
@gkc yes, there's a bunch of vestigial files. I'll use your branch and this PR to clear them out.

https://github.com/atsign-company/secondaries-scripts/blob/master/base/docker-compose.yml shows how the environment is set for the staging secondaries.

File: at_functional_test/lib/secondary/base/config/config.yaml [hunk 1]
```diff
@@ -9,8 +9,7 @@ root_server:
 
 # The @Protocol application log configurations
 log:
-  debug: true
-  trace: true
+  level: FINEST
```
Comment (cpswan) [inline at_functional_test/lib/secondary/base/config/config.yaml:12]: We might want to keep FINEST here for when we run tests.

File: at_secondary/at_secondary_server/config/config.yaml [hunk 2]
```diff
@@ -9,8 +9,7 @@ root_server:
 
 # The @Protocol application log configurations
 log:
-  debug: true
-  trace: true
+  level: FINEST
```
Comment (cpswan) [inline at_secondary/at_secondary_server/config/config.yaml]: INFO here I think

File: at_secondary/at_secondary_server/lib/src/server/at_secondary_config.dart [hunk 3]
```diff
@@ -59,9 +59,8 @@ class AtSecondaryConfig {
   static final int? _stats_top_keys = 5;
   static final int? _stats_top_visits = 5;
 
-  //log configurations
-  static final bool _debugLog = true;
-  static final bool _traceLog = true;
+  //log level configuration. Value should match the name of one of dart logging package's Level.LEVELS
+  static final String _defaultLogLevel = 'FINEST';
```
Comment (cpswan) [inline at_secondary/at_secondary_server/lib/src/server/at_secondary_config.dart]: INFO here too

**Review by cpswan — CHANGES_REQUESTED** [review]

File: at_functional_test/lib/secondary/base/config/config.yaml (hunk 1 above)
Comment (gkc) [reply at_functional_test/lib/secondary/base/config/config.yaml:12]: yes I think so

File: at_secondary/at_secondary_server/config/config.yaml (hunk 2 above)
Comment (gkc) [reply at_secondary/at_secondary_server/config/config.yaml]: Done

File: at_secondary/at_secondary_server/lib/src/server/at_secondary_config.dart (hunk 3 above)
Comment (gkc) [reply at_secondary/at_secondary_server/lib/src/server/at_secondary_config.dart]: Done

**Review by cpswan — APPROVED** [review]
LGTM

**Pull request merged**